cmd/tink-worker/main.go

```diff
@@ -22,35 +23,36 @@ func main() {
 
 	// Here we retry failed connections to tink-server using a randomized interval between attempts.
 	// tink-worker is a daemon process, so we do not exit here.
-	var conn *grpc.ClientConn
+	var creds credentials.TransportCredentials
 	for {
-		creds, err := tink.ObtainServerCreds(flagEnvSettings.TinkServerURL)
+		fmt.Printf("Obtaining tink server creds from %s...\n", flagEnvSettings.TinkServerURL)
+		var err error
+		creds, err = tink.ObtainServerCreds(flagEnvSettings.TinkServerURL)
 		if err == nil {
-			conn, err = tink.EstablishServerConnection(flagEnvSettings.TinkServerGRPCAuthority, creds)
-			if err == nil {
-				break
-			}
-			fmt.Printf("Error establishing gPRC connection to tink-server at %s: %v:", flagEnvSettings.TinkServerGRPCAuthority, err)
-		} else {
-			fmt.Printf("Error obtaining server creds from %s: %v", flagEnvSettings.TinkServerURL, err)
```

Comment:
If we elect to use the original code here (refer to my comment on `grpc.Dial`), I think we could simplify this `if` statement to be easier to read, and to provide more information to the operator by logging dial failures.

Reply:
I was thinking the connection loop might be a bit more readable if we try to reduce indentation. Something like this perhaps:

```go
sleepFn := func() {
	// sleep a randomized amount of time before reconnecting to avoid thundering herds
	rand.Seed(time.Now().UnixNano())
	s := rand.Intn(120) + 1 // 2 minutes (120 seconds), plus one second to avoid a possible zero sleep time
	fmt.Fprintf(os.Stderr, "Sleeping %d seconds before attempting to re-connect...\n", s)
	time.Sleep(time.Duration(s) * time.Second)
}

// Here we retry failed connections to tink-server using a randomized interval between attempts.
// tink-worker is a daemon process, so we do not exit here.
var conn *grpc.ClientConn
for {
	fmt.Fprintf(os.Stderr, "Obtaining tink server creds from %s...\n", flagEnvSettings.TinkServerURL)
	creds, err := tink.ObtainServerCreds(flagEnvSettings.TinkServerURL)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error obtaining server creds from %s: %v\n", flagEnvSettings.TinkServerURL, err)
		sleepFn()
		continue
	}

	fmt.Fprintf(os.Stderr, "Connecting to tink-server...\n")
	conn, err = tink.EstablishServerConnection(flagEnvSettings.TinkServerGRPCAuthority, creds)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error connecting to tink-server at %s: %v\n", flagEnvSettings.TinkServerGRPCAuthority, err)
		sleepFn()
		continue
	}

	break
}
```

What do you think?

Reply:
Yeah, I like this and have added it. Thanks!